perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_entityless_attribute_def_def_dnsStateMapping.java

```diff
@@ -0,0 +1,59 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.exceptions.*;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.EntitylessAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.EntitylessAttributesModuleImplApi;
+
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+/**
+ * Entityless attribute for mapping values of DNS address to specific state (or empty string).
+ *
+ * Examples:
+ * <BR>.cz = “Czech Republic”
+ * <BR>.uk = “United Kingdom”
+ * <BR>.ebi.ac.uk = “”
+ *
+ * @author Vladimir Mecko vladimir.mecko@gmail.com
+ */
+@SuppressWarnings("unchecked")
+public class urn_perun_entityless_attribute_def_def_dnsStateMapping extends EntitylessAttributesModuleAbstract implements EntitylessAttributesModuleImplApi {
+
+	static String KEY = "config";
+
+	public void checkAttributeValue(PerunSessionImpl perunSession, String key, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		if(attribute.getValue() == null) return;
+		LinkedHashMap<String, String> map = (LinkedHashMap<String, String>) attribute.getValue();
+		if(map.isEmpty()) return;
+
+		if(!KEY.equals(key)) {
```

Comment:
This check for key value should be before the check for null value and empty list. Please move the check to the first place among checks.